plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,467 @@
+
+    Encrypted strings are serialized as follows:
+    * padded using PKCS7 then encrypted using AES in CBC mode.  The AES key is
+      from the general.keystore file and the initialization vector is null.
+    * URL safe base64 encoded.
+
+    Args:
+      extracted_strings (list[str]): strings extracted from OneDrive log files.
+      aes_key (bytes): the AES key.
+
+    Returns:
+      list[str]: the decrypted strings.
+    """
+    if not aes_key:
+      return extracted_strings
+    output_strings = []
+
+    for extracted_string in extracted_strings:
+      tokens = self._DELIMITERS_REGEX.split(extracted_string)
+
+      output_string = []
+      for token in tokens:
+        if token in self._DELIMITERS:
+          output_string.append(token)
+          continue
+
+        token_length = len(token)
+
+        # minimum block size of base64 encoded AES-CBC is 22 bytes
+        if token_length < 22:
+          output_string.append(token)
+          continue
+
+        if not all(character in self._BASE64_ALPHABET for character in token):
+          output_string.append(token)
+          continue
+
+        rem = token_length % 4
+        if rem == 1:
+          output_string.append(token)
+          continue
+
+        while len(token) % 4:
+          token += '='
+
+        ciphertext = base64.urlsafe_b64decode(token)
+
+        if len(ciphertext) % 16:
+          output_string.append(token)
+          continue
+
+        aes_context = pycaes.context()
+        aes_context.set_key(pycaes.crypt_modes.DECRYPT, aes_key)
+        plaintext = pycaes.crypt_cbc(
+            aes_context, pycaes.crypt_modes.DECRYPT,
+            self._AES_IV, ciphertext)
+
+        unpadder = padding.PKCS7(128).unpadder()
+        try:
+          data = unpadder.update(plaintext) + unpadder.finalize()
+        except ValueError:
+          output_string.append(token)
+          continue
+
+        try:
+          output_string.append(data.decode('utf-16-le'))
+        except UnicodeError:
+          output_string.append(data)
+
+      output_strings.append(''.join(output_string))
+    return output_strings
+
+  def _ExtractStringsFromParameters(self, buffer):
+    """Extracts strings from raw parameter bytes.
+
+    Args:
+      buffer (bytearray): the buffer.
+      aes_key (bytes): the AES decryption key from general.keystore.
+      obfuscated_string_map (dict[str, str]): the mapping of obfuscated string
+          values to their original values.
+
+    Returns:
+      list[str]: strings parsed from the raw parameter bytes.
+    """
+    pascal_string_map = self._GetDataTypeMap('pascal_string')
+
+    parse_offset = 0
+    string_array = []
+
+    while parse_offset < len(buffer):
+      try:
+        pascal_string = self._ReadStructureFromByteStream(
+            buffer[parse_offset:], 0, pascal_string_map)
+
+        if len(pascal_string.value) and all(
+            c < 128 for c in pascal_string.value):
```

Comment:
nit: don't use abbreviations what is `c` ?

Reply:
Done